lib/src/public_api_parser.dart

```diff
@@ -36,10 +36,26 @@ class InsertionOrderParser {
       key: (v) => v.name,
       value: (v) => v);
 
-  /// Parses a json string to [InsertionOrder].
-  static InsertionOrder parse(String jsonString) {
+  /// Parses a list of [InsertionOrder] from a [jsonString].
+  ///
+  /// Returns an empty list if [jsonString] is null or empty.
+  static List<InsertionOrder> parseInsertionOrders(String jsonString) {
+    if (jsonString == null || jsonString.isEmpty) return [];
     Map<String, dynamic> map = json.decode(jsonString);
-    return _createInsertionOrder(map);
+    return map.containsKey('insertionOrders')
+        ? List.from(map['insertionOrders'])
```

Comment:
meta comment this is about as big as a ternary should get from a personal readability preference.